Externalize auth providers (#1055)
SAP/luigi: plugins/public/auth-oidc/README.md

# OpenID Connect - Authorization Plugin for Luigi Core

## Overview

> This project contains a library that allows your application to extend the [Luigi framework](https://github.com/SAP/luigi/tree/master/core) with a OpenID Connect authorization provider. 

Reviewer: ```suggestion
This project contains a library that allows your application to extend the [Luigi framework](https://github.com/SAP/luigi/tree/master/core) with an OpenID Connect authorization provider. 
```

> Further configuration details can be found in the main documentation: [authorization-configuration](https://docs.luigi-project.io/docs/authorization-configuration#oauth2-implicit-grant-configuration)

Reviewer: ```suggestion
Further configuration details can be found in the [main documentation](https://docs.luigi-project.io/docs/authorization-configuration#openid-connect-configuration).
```

## Installation

> Install the client in your project using npm:

Reviewer: ```suggestion
Install the plugin in your project using npm:
```

```bash
npm install @luigi-project/plugin-auth-oidc
```

> Import the client in places where you want to use it, depending on the environment of your choice:

Reviewer: ```suggestion
Import the plugin in places where you want to use it, depending on the environment of your choice:
```

```javascript
var OpenIdConnect = require('@luigi-project/plugin-auth-oidc');
```
or
```javascript
import OpenIdConnect from '@luigi-project/plugin-auth-oidc';
```

> Then integrating it as authorization provider:

Reviewer: ```suggestion
Then, integrate it as an authorization provider in your Luigi configuration file:
```

```javascript
Luigi.setConfig({
  auth: {
    use: 'myProviderConfig',
    myProviderConfig: {
      idpProvider: OpenIdConnect,
      authority: 'http://authority.server',
      logoutUrl: 'http://authority.server/connect/endsession',
      client_id: 'implicit-mock-client',
      scope: 'openid profile email',
      // ... further configuration data comes here
    }
  }
})
```

> If you want to use the silent token renewal feature, the silent-callback.html needs to be copied to a folder in Luigi Core installation, which is the return path for idp provider, configured through the `redirect_uri` setting. Default location of `redirect_uri` is `/assets/auth-oidc/silent-callback.html`. This file contains also a reference to `/assets/auth-oidc/plugin.js` to make the OIDC client available.

Reviewer: ```suggestion
If you want to use the silent token renewal feature, the `silent-callback.html` needs to be copied to a folder in your Luigi Core installation, which is the return path for the IdP provider, configured through the `redirect_uri` setting. The default location of `redirect_uri` is `/assets/auth-oidc/silent-callback.html`. This file also contains a reference to `/assets/auth-oidc/plugin.js` to make the OIDC client available.
```
